### ISSUE TITLE
Error compiling flexible_jobshop.cc from VS2017 binary

### ISSUE THREAD

**Issue by aldeo** [google/or-tools#709]
Hi, 
I downloaded the or-tools binary for VS2017 from [https://github.com/google/or-tools/releases/download/v6.7.1/or-tools_VisualStudio2017-64bit_v6.7.4973.zip](url) and got the following error while compiling the examples: 
```
examples\cpp\flexible_jobshop.cc(228): error C2039: 'Join': is not a member of 'strings'
include\ortools/base/stringpiece_utils.h(21): note: see declaration of 'strings'
examples\cpp\flexible_jobshop.cc(227): error C3861: 'Join': identifier not found
make: *** [objs\\flexible_jobshop.obj] Error 2
```

**Steps to reproduce:**
- Open x64 Native Tools Command Prompt for VS 2017
- Go to directory where .zip file was extracted
- Run "make all"

**How I fixed it locally:**
- Open file ""..\examples\cpp\flexible_jobshop.cc""
- Add header file: `#include "ortools/base/join.h"`
- Replace `string::Join` -> `absl::StrJoin `in line 229

This fixes the compilation issue but I wonder whether it is the right way to do it.

**Comment by lperron** [comment]
Thanks, I did not fix this example after all the changes in the base
library.
I will have a look.

Please note that using jobshop_sat.cc along the
ortools/data/jobshop_scheduling.proto is recommended.

--Laurent
Laurent Perron | Operations Research | lperron@google.com | (33) 1 42 68 53
00

**Comment by lperron** [comment]
Please note that if you compile from source, the flexible_jobshop compiles
fine.

Thanks
Laurent Perron | Operations Research | lperron@google.com | (33) 1 42 68 53
00

**Comment by Mizux** [comment]
I've created a `6.7.x` branch I'll backport the patch 023bc19 and recreate an archive on windows will be tagged "v6.7.2"

**Comment by jkornblum** [comment]
Sounds like this error may be fixed already but not merged into master but FYI

I'm also getting this same error building my docker container from `tiangolo/uwsgi-nginx-flask:python3.6` which is  built from `debian:sid`. After `make all` my exact error is 

```
examples/cpp/flexible_jobshop.cc: In function void operations_research::FlexibleJobshop(const operations_research::FlexibleJobShopData&):
examples/cpp/flexible_jobshop.cc:228:20: error: ‘Join’ is not a member of ‘strings’
                 << strings::Join(collector->ForwardSequence(0, seq), ", ");
```
My full Dockerfile is this for reference

```
FROM tiangolo/uwsgi-nginx-flask:python3.6

RUN set -ex; \
	apt-get update; \
	apt-get install -y --no-install-recommends \
		build-essential;

COPY or-tools_Debian-9.3-64bit_v6.7.4973.tar.gz /opt

RUN cd /opt; \
    tar -zxf /opt/or-tools_Debian-9.3-64bit_v6.7.4973.tar.gz; \
    rm /opt/or-tools_Debian-9.3-64bit_v6.7.4973.tar.gz; \
    mv /opt/or-tools_Debian-9.3-64bit_v6.7.4973 /opt/google-or-tools; \    
    cd /opt/google-or-tools; \
    make all; \
    python -m pip install --upgrade ortools; \
    python -m pip show ortools;

ENV STATIC_INDEX 1

COPY ./app /app
```

**Comment by Mizux** [comment]
In fact it's already fix on master (for 6.8) but not backported to branch 6.7.x (just created).

Waiting for CI to pass them hope to release new package or-tools_Debian-9.3-64bit_v6.7.4978.tar.gz on Monday, tag will be v6.7.2

note: python ortools is standalone, i.e. contains all needed libs -> no need of the tar.gz for python usage

**Comment by jkornblum** [comment]
Haha, yes that is clear as day in the documentation now that I read it again :smile: 

Cheers.
